src/qt/upgradeqt.cpp

```diff
@@ -341,3 +358,39 @@ void UpgradeQt::DeleteSnapshot()
         LogPrintf("Snapshot Downloader: Exception occurred while attempting to delete snapshot (%s)", e.code().message());
     }
 }
+
+bool UpgradeQt::SyncFromZero(QApplication& SyncfromzeroApp)
+{
+    SyncfromzeroApp.processEvents();
+    SyncfromzeroApp.setWindowIcon(QPixmap(":/images/gridcoin"));
+
+    Upgrade syncfromzero;
+
+    bool fSuccess = syncfromzero.CleanupBlockchainData();
+
+    if (fSuccess)
+        Msg(_("Sync from zero: Blockchain data removal was a Success"), _("The wallet will now shutdown. Please start your wallet to begin sync from zero"), false);
+
+    else
+    {
+        std::string inftext = "";
+        // Little more work then needed but we should be translation friendly imo
+        inftext.append(_("Datadir: "));
+        inftext.append(GetDataDir().string());
+        inftext.append("\r\n\r\n");
+        inftext.append(_("Due to the failure to delete the blockchain data you will be required to manually delete the data before starting your wallet."));
+        inftext.append("\r\n");
+        inftext.append(_("Failure to do so will result in undefined behaviour or failure to start wallet."));
+        inftext.append("\r\n\r\n");
+        inftext.append(_("You will need to delete the following."));
+        inftext.append("\r\n\r\n");
+        inftext.append(_("Files:"));
+        inftext.append("\r\nblk000*.dat\r\n\r\n");
+        inftext.append(_("Directories:"));
+        inftext.append("\r\ntxleveldb\r\naccrual");
```

Comment:
Can we wrap this block of static text construction into a function? Someone may forget to update all three places that use the same text in the future.

Reply:
this has been done in upgrade.cpp/h